10.0 porting website_sale_checkout_skip_payment

eLBati opened the pull request.

sergio-teruel committed:
[9.0][WIP] website_sale_checkout_skip_payment: New module to skip payment step in checkout process

chienandalu committed:
[IMP] website_sale_checkout_skip_payment: Texts + code

oca-transbot committed:
OCA Transbot updated translations from Transifex

OCA Transbot updated translations from Transifex

eLBati committed:
Porting website_sale_checkout_skip_payment to 10.0

tafaRU committed:
[9.0][FIX] website_sale_checkout_skip_paymen: Message error in checkout confirmation step when partner is has skip payment flag (#1)

simahawk commented on website_sale_checkout_skip_payment/README.rst:
```diff
@@ -0,0 +1,63 @@
+Website Sale Checkout Skip Payment
+==================================
+
+This module allows to logged users to checkout with no payment step. At the
```
"allows logged users"

eLBati replied:
@simahawk done

simahawk commented on website_sale_checkout_skip_payment/README.rst:
```diff
@@ -0,0 +1,63 @@
+.. image:: https://img.shields.io/badge/license-LGPL--3-blue.svg
```
png

eLBati replied:
@simahawk readme files re-created with new fragments

simahawk commented on website_sale_checkout_skip_payment/__manifest__.py:
```diff
@@ -0,0 +1,24 @@
+{
+    'name': 'Website Sale Checkout Skip Payment',
+    'summary': 'Skip payment for logged users in checkout process',
+    'version': '10.0.0.1.0',
```
10.0.1.0.0

eLBati replied:
@simahawk done

simahawk commented on website_sale_checkout_skip_payment/__manifest__.py:
```diff
@@ -0,0 +1,24 @@
+    'summary': 'Skip payment for logged users in checkout process',
+    'version': '10.0.0.1.0',
+    'category': 'Website',
+    'website': 'https://www.tecnativa.com',
```
https://github.com/OCA/e-commerce

eLBati replied:
@simahawk done

simahawk commented on website_sale_checkout_skip_payment/controllers/main.py:
```diff
@@ -0,0 +1,37 @@
+
+    @http.route()
+    def payment_get_status(self, sale_order_id, **post):
+        # When skip payment step, the transaction not exists so only render
```
transaction does not exists

eLBati replied:
@simahawk done

simahawk commented on website_sale_checkout_skip_payment/models/website.py:
```diff
@@ -0,0 +1,20 @@
+    @api.multi
+    def _compute_checkout_skip_payment(self):
+        for rec in self:
+            if request.session.uid:
```
just checking: is 100% that this is going to be recomputed for each user?
I'd rather use a std model method to get it not cached.

eLBati replied:
@simahawk just tested with 2 different users on 2 different browsers: it works

grindtildeath commented:
I get this error when confirming an order through web shop :
```
2018-04-27 09:50:45,531 1 ERROR bssfw-24 odoo.http: Exception during JSON request handling.
Traceback (most recent call last):
  File "/opt/odoo/src/odoo/http.py", line 640, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/opt/odoo/src/odoo/http.py", line 677, in dispatch
    result = self._call_function(**self.params)
  File "/opt/odoo/src/odoo/http.py", line 333, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/opt/odoo/src/odoo/service/model.py", line 101, in wrapper
    return f(dbname, *args, **kwargs)
  File "/opt/odoo/src/odoo/http.py", line 326, in checked_call
    result = self.endpoint(*a, **kw)
  File "/opt/odoo/src/odoo/http.py", line 935, in __call__
    return self.method(*args, **kw)
  File "/opt/odoo/src/odoo/http.py", line 506, in response_wrap
    response = f(*args, **kw)
  File "/opt/odoo/external-src/e-commerce/website_sale_checkout_skip_payment/controllers/main.py", line 35, in payment_get_status
    'message': request.website._render(
AttributeError: 'website' object has no attribute '_render'
```

rafaelbn commented:
Hi @eLBati thanks for this migration! Please could you check comments? Thanks

rafaelbn commented:
Hello @eLBati could you review comments? Are you interested in continuing this PR?

- The v11 migration is here https://github.com/OCA/e-commerce/pull/228
- We did it duplicated in https://github.com/OCA/e-commerce/pull/246 :yin_yang: 
- You should take care of https://github.com/OCA/e-commerce/pull/223
- Check comments from @simahawk 

Let me know! :smiley:

eLBati commented:
@grindtildeath can you describe the steps to reproduce the error? I am not able to reproduce it in a standard scenario

eLBati commented:
@rafaelbn thanks, https://github.com/OCA/e-commerce/pull/223 is already included with https://github.com/OCA/e-commerce/pull/203/commits/e26b2c5da1c34eaacb5fb88bf59b9f9026f7b8d8

eLBati committed:
IMP README files and meta data

FIX
website_sale_checkout_skip_payment/controllers/main.py", line 35, in payment_get_status
    'message': request.website._render(
AttributeError: 'website' object has no attribute '_render'

grindtildeath reviewed (commented):
@eLBati Tested again and seems to work, so approved.

grindtildeath commented:
@eLBati My bad, but while switching off my local instances, I just see I had the same error in my server logs, I'll try to give it another look tomorrow.

grindtildeath commented:
@eLBati Just tested again and this happens when confirming the order on the webshop with a user having skip payment activated. The UX looks ok, but we get this error in the logs.

eLBati commented:
@grindtildeath thanks, fixed

fuentes73 reviewed (approved):
Using it in production since a while and everything works OK.

rafaelbn commented:
/ocabot merge

zeroheure commented:
Why this was never merged ?
